clippy_lints/src/utils/mod.rs

```diff
@@ -302,7 +302,7 @@ pub fn implements_trait<'a, 'tcx>(
         cx.tcx
             .predicate_for_trait_def(cx.param_env, traits::ObligationCause::dummy(), trait_id, 0, ty, ty_params);
     cx.tcx.infer_ctxt().enter(|infcx| {
-        traits::SelectionContext::new(&infcx).evaluate_obligation_conservatively(&obligation)
+        traits::SelectionContext::new(&infcx).infcx().predicate_must_hold(&obligation)
```

Comment:
Sorry, didn't see this PR earlier but this could just be `infcx.predicate_must_hold(...)` --- i.e., no need to create a `SelectionContext`. No functional difference though, and I suppose not much perf difference in practice either, so not a big deal.